pynest/examples/compartmental_model/receptors_and_current.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+#
+# receptors_and_current.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Example of a passive three-compartment model that is connected to spike-
+generators via three receptors types and is also connected to a current
+generator.
+"""
+
+import nest
+import matplotlib.pyplot as plt
+
+nest.ResetKernel()
+nest.SetKernelStatus(dict(resolution=.1))
+
+# somatic and dendritic parameters
+soma_params = {
+    'C_m': 10.0,    # [pF] Capacitance
+    'g_C': 0.0,     # soma has no parent
+    'g_L': 1.,      # [nS] Leak conductance
+    'e_L': -70.0    # [mV] leak reversal
+}
+dend_params = {
+    'C_m': 0.1,     # [pF] Capacitance
+    'g_C': 0.1,     # [nS] Coupling conductance to parent (soma here)
+    'g_L': 0.1,     # [nS] Leak conductance
+    'e_L': -70.0    # [mV] leak reversal
+}
+
+# create a model with three compartments
+cm = nest.Create('cm_default')
+nest.SetStatus(cm, {'compartments': {"parent_idx": -1, "params": soma_params}})
```

Comment:
yes, this works, although I find the syntax somewhat counterintuitive if you set the compartment structure through multiple of such lines, e.g.
```
cm.compartments = {"parent_idx": -1, "params": soma_params}
cm.compartments = {"parent_idx": 0, "params": dend_params}
```
however, in a single call, it looks nice
```
cm.compartments = [
    {"parent_idx": -1, "params": soma_params},
    {"parent_idx": 0, "params": dend_params}
]
```

Reply:
You're right.

Maybe some day we can expose a model-specific `+=` operator to Python...

However, this makes me think that the documentation of the model definitely needs to be very about this peculiarity and the fact that consecutive calls to `SetStatus()` or `=` will indeed not overwrite the set of existing compartments/receptors, but actually *add* to the list of already existing ones.

Reply:
The fact that `cm.compartments = { ... }` adds new compartments instead of setting all compartments dangerous, since it is not intuitive. Is there a very good reason to build the neuron over several lines instead of building a dictionary specifying the entire neuron first and then passing it to the neuron in a single call? Trying to set compartments on a neuron that already has compartments should then throw an error.

Reply:
This is mostly leftover from the original implementation, where compartments could only be added one by one.

However, a good reason is easier debugging. If the compartments are added one by one, and something with the tree structure is wrong (e.g. a `parent_idx` that is not in the tree), then you will get an error at the specific line in the python script where you tried to add the compartment. If they are all in one big list, then you have to sift through the list trying to find where you added the compartment with the incorrect parent (an error with the relevant information will be raised, but you would still have to look somehow for the element in the list). I would support the `+=` idea. This would also implement the option to add receptors between different `nest.Simulate()` calls.

Reply:
Would this use-case warrant the addition of a a model-specific `+=` operator on the level of the Python wrapper for `NodeCollection`s? @stinebuu

In any case, I think we should merge this PR asap and only advertise the *set-all-compartments-at-once* use-case in the documentation (maybe even prohibit further calls to `=`/`SetStatus`?), create an issue with all the information collected here, and then add new functionality in a follow-up PR. @heplesser, if you agree with this course of action, please let @WillemWybo know, so he can create a corresponding issue.

Reply:
I agree that we should merge this asap. Trying to set the `compartments` twice shall raise an error now (should be straightforward to implement at the C++-level, I think) to avoid that users write scripts with subtle errors. Then, in a later stage (new issue, please), we can add more informative error messages (when iterating over the compartment list, and error message should tell which entry in the list caused the problem) and/or the `+=` operator.

Reply:
I have disabled the possibility to set compartments or receptors twice. I'll create an issue on the `+=` once the PR has been merged.

Reply:
LGTM for now, thus resolving. Please don't forget to link [this conversation](https://github.com/nest/nest-simulator/pull/1895#discussion_r805829211) directly in the new issue.